doc/MAPGEN.md

```diff
@@ -1386,3 +1389,143 @@ update_mapgen adds new optional keywords to a few mapgen JSON items.
+| `mapgen`             | The `generator_id` points to a `om_terrain` string within a `mapgen` definition.
+| `update_mapgen`      | The `generator_id` points to a `update_mapgen_id` string within a `mapgen` definition.
+
+### Example: mx_science
+
+The `mx_science` map extra spawns bodies of scientists as well as a few enemy mobs. The map extra definition is as follows:
+
+```JSON
+{
+  "id": "mx_science",
+  "type": "map_extra",
+  "name": { "str": "Scientists" },
+  "description": "Several corpses of scientists are here.",
+  "generator": { "generator_method": "update_mapgen", "generator_id": "mx_science" },
+  "min_max_zlevel": [ -5, 0 ],
+  "sym": "s",
+  "color": "light_red",
+  "autonote": true
+}
+```
+
+In this case the `generator_id` points to a `mapgen` definition that establishes what objects or structures will spawn at that location:
+
+```JSON
+{
+  "type": "mapgen",
+  "method": "json",
+  "update_mapgen_id": "mx_science",
+  "object": {
+    "rows": [
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "-----------1----------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "----------------------  ",
+      "                        ",
+      "                        "
+    ],
+    "monsters": {
+      " ": { "monster": "GROUP_NETHER_CAPTURED", "chance": 1, "density": 0.0001 },
+      "-": { "monster": "GROUP_NETHER_CAPTURED", "chance": 1, "density": 0.0001 }
+    },
+    "nested": {
+      "-": { "chunks": [ [ "corpse_blood_gibs_science_3x3", 1 ], [ "null", 150 ] ] },
+      "1": { "chunks": [ "corpse_blood_gibs_science_3x3" ] }
+    }
+  }
+}
+```
+
+The nested chunks define the items and fields that may spawn:
+
+```JSON
+{
+  "type": "mapgen",
+  "method": "json",
+  "nested_mapgen_id": "corpse_blood_gibs_science_3x3",
+  "object": {
+    "mapgensize": [ 3, 3 ],
+    "place_items": [ { "item": "map_extra_science", "x": [ 0, 2 ], "y": [ 0, 2 ], "chance": 100 } ],
+    "place_fields": [ { "field": "fd_blood", "x": [ 0, 2 ], "y": [ 0, 2 ] }, { "field": "fd_gibs_flesh", "x": [ 0, 2 ], "y": [ 0, 2 ] } ]
+  }
+}
+```
+
+In order for the map extra to be available for spawning, one or more `region_settings` entries need to be created for it:
+
+```JSON
+{
+  "type": "region_settings",
+  "id": "default",
+  ...
+  "map_extras": {
+    "forest": {
+      "chance": 20,
```

Comment:
It would be nice to explain what this chance line does. I vaguely remember it's X in 1000? I think that's true for the numbers after the map extra ID